tests/integration/test_api_gateway.py

```diff
@@ -434,6 +434,26 @@ def test_apigateway_with_lambda_integration(self):
             restApiId=api_id
         )
 
+    def test_api_gateway_handle_domain_name(self):
+        domain_name = '%s.localstack.com' % short_uid()
```

Comment:
nit: Can we change this to a synthetic domain name, to avoid confusion (`localstack.com` is an existing domain, but not related to us):
```
domain_name = '%s.example.com' % short_uid()
```